### PR TITLE
Get user from token as required for magic login to work

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes # (issue)

**Steps to replicate the issue:**
- Make sure that you are logged out from app.cal.com
- Go to cal.com/signup/verify?email=YOUR_CAL_EMAIL
- Click on Button to receive a verification email
- Click on the link provided in the verification email 
- Notice that the page would be hung for a long time and then it would show
image

The issue wasn't happening on local because local setup is considered as !hosted. @deepakprabhakara Need your help to figure out why this auto merge identities change was required only on !hosted as I have open it up for cal.com as well now.

**Environment**: Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Logged in with Credentials
- [ ]  Logged in from Verification Email Link(Magic Link)

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 1]
```diff
@@ -215,29 +215,26 @@ export default NextAuth({
...
+        return {
+          id: existingUser.id,
+          username: existingUser.username,
+          name: existingUser.name,
+          email: existingUser.email,
+          role: existingUser.role,
+          impersonatedByUID: token?.impersonatedByUID as number,
+        };
+
         return token;
```
Comment (sean-brydon) [inline apps/web/pages/api/auth/[...nextauth].tsx:236]:
```suggestion
```

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 2]
```diff
@@ -215,29 +215,26 @@ export default NextAuth({
...
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            where: { email: token.email! },
-          });
-
-          if (!existingUser) {
-            return token;
-          }
+        const existingUser = await prisma.user.findFirst({
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          where: { email: token.email! },
```
Comment (sean-brydon) [inline apps/web/pages/api/auth/[...nextauth].tsx:220]: Could we parse this with zod to ensure this is typesafe? Haven't tested but something like this should stop eslint/TS complaining here 

CleanShot 2022-09-05 at 13 40 02@2x

**Review by sean-brydon — DISMISSED** [review]

File: apps/web/pages/api/auth/[...nextauth].tsx (hunk 2 above)
Comment (zomars) [reply apps/web/pages/api/auth/[...nextauth].tsx:220]: That can be done in a follow up

**Review by zomars — APPROVED** [review]
Very nice find @hariombalhara this is a reminder that we should be extra careful when introducing `cal.com` only logic.

**Comment by agustif** [comment]
Great work on this one @hariombalhara Thank you!!

**Pull request merged**